Project/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -9,14 +9,26 @@
 
 public class WalkerAgent : Agent
 {
-    public float maximumWalkingSpeed = 999; //The max walk velocity magnitude an agent will be rewarded for
-    Vector3 m_WalkDir; //Direction to the target
-//    Quaternion m_WalkDirLookRot; //Will hold the rotation to our target
+    [Header("Walk Speed")] [Range(0, 10)] public float walkingSpeed = 10; //The walking speed to try and achieve
+    float m_maxWalkingSpeed = 10; //The max walking speed
 
-    [Header("Target To Walk Towards")] [Space(10)]
-    public TargetController target; //Target the agent will walk towards.
+    //Should the agent sample a new goal velocity each episode?
+    //If true, walkSpeed will be randomly set between zero and m_maxWalkingSpeed in OnEpisodeBegin() 
+    //If false, the goal velocity will be walkingSpeed
+    public bool randomizeWalkSpeedEachEpisode;
 
-    [Header("Body Parts")] [Space(10)] public Transform hips;
+    public enum WalkDirectionMethod
+    {
+        UseWorldDirection,
+        UseTarget
+    }
+
+    [Header("Walk Direction")] public WalkDirectionMethod walkDirectionMethod;
+    public Vector3 worldDirToWalk = Vector3.right;
+    public Vector3 worldPosToWalkTo;
+    public Transform target; //Target the agent will walk towards.
```

Comment:
I tried to modify these values in the inspector but I don't seem to be able to change them at runtime. I think it will confuse users.

Reply:
Added more code comments. agreed it's confusing but we don't have the ability to conditionally hide properties in the editor.  Only WalkerStatic and WalkerStaticVariable speed use the UseWorldDirection walk method. If you change the worldDirToWalk in those scenes, they will run towards the specified direction. 

This design with multiple WalkDirectionMethods was necessary so that an agent could be given either a target to run towards in the case of NPCs or given a world direction which could be used for player input.